### PR TITLE
feat(DTFS2-6862): add cron job to delete acbs logs

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
we needed to add a cron job to clean acbs durable function logs every day at midnight

## Resolution :heavy_check_mark:
added a cron job to delete acbs durable function logs every day at midnight

## Miscellaneous :heavy_plus_sign:
there's some pluralisation that's a bit surprising but I've kept this pattern

### CODE REVIEW

File: dtfs-central-api/src/cron-scheduler-jobs/delete-acbs-durable-function-logs/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('calls deleteAllAcbsDurableFunctionLogs from the durable functions repo', async () => {
+      // Arrange
+      jest.mocked(deleteAllAcbsDurableFunctionLogs).mockResolvedValue({ acknowledged: true, deletedCount: 1 });
+
+      // Act
+      await deleteAcbsDurableFunctionLogsJob.task(new Date());
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/cron-scheduler-jobs/delete-acbs-durable-function-logs/index.test.ts:30]: I'm a bit confused why this needs an argument passed in

File: dtfs-central-api/src/services/repositories/durable-functions-repo.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Collection, DeleteResult } from 'mongodb';
+import db from '../../drivers/db-client';
+import { DB_COLLECTIONS } from '../../constants';
+
+const getDurableFunctionsLogCollection = async (): Promise<Collection> => db.getCollection(DB_COLLECTIONS.DURABLE_FUNCTIONS_LOG);
+
+export const deleteAllDurableFunctionLogs = async (): Promise<DeleteResult> => {
+  const durableFunctionLogsCollection = await getDurableFunctionsLogCollection();
+  return await durableFunctionLogsCollection.deleteMany({});
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/services/repositories/durable-functions-repo.ts:9]: Think the await is unnecessary & the type `Promise<DeleteResult>` will be inferred so doesn't need to be stated. Happy to leave if you think improves readability

Comment (MarRobSoftwire) [reply dtfs-central-api/src/services/repositories/durable-functions-repo.ts:9]: same for line 14 too

File: dtfs-central-api/src/services/repositories/durable-functions-repo.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+};
+
+export const deleteAllAcbsDurableFunctionLogs = async (): Promise<DeleteResult> => {
+  const durableFunctionLogsCollection = await getDurableFunctionsLogCollection();
+  return await durableFunctionLogsCollection.deleteMany({ type: { $eq: 'ACBS' } });
```
Comment (MarRobSoftwire) [inline dtfs-central-api/src/services/repositories/durable-functions-repo.ts:14]: Should `'ACBS'` be a constant?

File: dtfs-central-api/src/services/repositories/durable-functions-repo.ts (hunk 2 above)
Comment (AlexBramhill) [reply dtfs-central-api/src/services/repositories/durable-functions-repo.ts:9]: Very true

File: dtfs-central-api/src/services/repositories/durable-functions-repo.ts (hunk 3 above)
Comment (AlexBramhill) [reply dtfs-central-api/src/services/repositories/durable-functions-repo.ts:14]: added

File: dtfs-central-api/src/cron-scheduler-jobs/delete-acbs-durable-function-logs/index.test.ts (hunk 1 above)
Comment (AlexBramhill) [reply dtfs-central-api/src/cron-scheduler-jobs/delete-acbs-durable-function-logs/index.test.ts:30]: So, this is how it's going to actually be called by node-cron -- I agree its a bit odd to see in a unit test but this is the existing pattern.

This task eventurally gets passed into node.schedule, which then gets called with `date`

Hope this helps -- Took a bit of digging to find (and a chat with Josh to speed things along!)

Comment (AlexBramhill) [reply dtfs-central-api/src/cron-scheduler-jobs/delete-acbs-durable-function-logs/index.test.ts:30]: I've added a comment on the type for future reference

File: trade-finance-manager-api/src/constants/index.js
Comment (MarRobSoftwire) [inline trade-finance-manager-api/src/constants/index.js:1]: Thanks for doing this commonising :)

File: trade-finance-manager-api/src/v1/controllers/acbs.controller.js [hunk 4]
```diff
@@ -106,19 +107,20 @@ const updateAmendedFacilityAcbs = (taskResult) => {
...
     const collection = await db.getCollection('durable-functions-log');
-    const runningTasks = await collection.find({
-      type: { $eq: 'ACBS' },
-      status: { $eq: 'Running' },
-    }).toArray();
-    const tasks = await runningTasks.map(({ acbsTaskLinks = {} }) =>
-      api.getFunctionsAPI(acbsTaskLinks.statusQueryGetUri));
+    const runningTasks = await collection
+      .find({
+        type: { $eq: 'ACBS' },
```
Comment (MarRobSoftwire) [inline trade-finance-manager-api/src/v1/controllers/acbs.controller.js:114]:
```suggestion
        type: { $eq: DURABLE_FUNCTIONS_LOG.TYPE.ACBS },
```

**Review by MarRobSoftwire — DISMISSED** [review]
Just a minor thing I noticed because of prettier

**Review by MarRobSoftwire — DISMISSED** [review]

**Comment by AlexBramhill** [comment]
closing this pr in attempt to get tests running

**Pull request closed without merging**